plugins/modules/mount.py

```diff
@@ -659,6 +710,25 @@ def get_linux_mounts(module, mntinfo_file="/proc/self/mountinfo"):
     return mounts
 
 
+def _is_same_mount_src(module, args, mntinfo_file="/proc/self/mountinfo"):
+    """Return True if the mounted fs on mountpoint is the same source than src"""
+    mountpoint = args['name']
+    src = args['src']
+    lines = _get_mount_info(module)
+
+    # If this function is used and we cannot retrieve mount info, we must fail to avoid unexpected behavior
+    if lines is None:
+        module.fail_json(msg="Unable to retrieve mount info from '%s'" % mntinfo_file)
+
+    for line in lines:
+        fields = line.split()
+        if fields[4] == mountpoint and fields[-2] == src:
+            return True
```

Comment:
`fields[4]` contains a mountpoint and `fields[-2]` contains the associated source. The logic behind picking these fields was already implemented in `get_linux_mounts`, I only copy-pasted after test and review.